apps/src/applab/CrosshairOverlay.js

```diff
@@ -140,3 +180,23 @@ CrosshairOverlay.prototype.getCoordinateText = function () {
   return "x: " + Math.floor(this.props_.x) +
       ", y: " + Math.floor(this.props_.y);
 };
+
+/**
+ * Internal helper to generate the element id string to display in tooltip.
+ * Since ids can be very long, this method also ellipsifies the id as needed.
+ * @returns {string}
+ * @private
+ */
+CrosshairOverlay.prototype.getElementIdText_ = function () {
+  if (!this.props_.mouseoverApplabControlId) {
+    return '';
+  }
+
+  // Check the length of the id and ellipsify as needed
+  var elementIdText = this.props_.mouseoverApplabControlId;
+  if (elementIdText.length > ELEMENT_ID_TEXT_MAX_CHAR) {
+    elementIdText = elementIdText.substr(0, ELEMENT_ID_TEXT_MAX_CHAR) + "...";
+  }
```

Comment:
This is mostly perfect so I'm going to get picky :grin:

I'd like to see this ellipsify method use available space better - right now, adding an ellipsis to the end of the string will cause the final string to be longer than your specified max length.  Here's the current behavior:

```
(9)  ABCDEFGHI    => ABCDEFGHI     (9)
(10) ABCDEFGHIJ   => ABCDEFGHIJ    (10)
(11) ABCDEFGHIJK  => ABCDEFGHIJ... (13)
(12) ABCDEFGHIJKL => ABCDEFGHIJ... (13)
```

I'd like to see the output stay within our specified length, but not ellipsify if the word fits.  Something like this:

```
(9)  ABCDEFGHI    => ABCDEFGHI  (9)
(10) ABCDEFGHIJ   => ABCDEFGHIJ (10)
(11) ABCDEFGHIJK  => ABCDEFG... (10)
(12) ABCDEFGHIJKL => ABCDEFG... (10)
```

Then, assuming the 13-character ellipsified strings were fitting okay in the UI before, you can safely increase your max char constant to 13 and show longer IDs without ellipses. I _think_ this would be closer to the behavior of the CSS3 `text-overflow: ellipsis;` property.

I know this is a trivial bit of logic, but it would be a perfect place for [functional core testing](https://www.destroyallsoftware.com/talks/boundaries).  Here's what you do:

1. Extract a static helper `CrosshairOverlay.ellipsifyText = function (inputText, maxLength) {...}`
2. Write your unit tests against that static method.  Since it's a [pure function](https://en.wikipedia.org/wiki/Pure_function) testing it won't require any special set-up, fakes, etc.